Purchases/Classes/Public/RCPurchases.m

```diff
@@ -41,16 +41,60 @@ @interface RCPurchases () <RCStoreKitWrapperDelegate>
 
 @implementation RCPurchases
 
+static RCPurchases *_singleton = nil;
+
+@synthesize delegate=_delegate;
+
++ (NSString *)frameworkVersion {
+    return @"1.2.0-SNAPSHOT";
+}
+
++ (instancetype)defaultInstance {
+    if (!_singleton) {
+        RCLog(@"There is no default instance. Make sure you configure Purchases before trying to get the default instance.");
+    }
+    return _singleton;
+}
+
++ (void)setDefaultInstance:(RCPurchases *)instance {
+    @synchronized(_singleton) {
```

Comment:
I wouldn't use `@synchronized` on the `_singleton` instance if that is going to be overwritten. Perhaps synchronize on `[RCPurchases class]`?

Reply:
Haha, I think we had that originally but I told him to change it. So I guess @synchronized works on the instance and not the variable? Makes sense.

I'm not used to using it in a static method.

Reply:
Correct, `@synchronized` works on the pointer itself essentially.